File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    let answer = 0;
    
    let currentItem = 0;
    
    section.forEach((item) => {
        if (item > currentItem) {
            currentItem = item + m - 1;
            answer++;
        }
    })
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 1
    
    current_value = section[0] + m - 1
    section.pop(0)
    
    for item in section:
        if item > current_value:
            answer += 1
            current_value = item + m - 1
    
    return answer